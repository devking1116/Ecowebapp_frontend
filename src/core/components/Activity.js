/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Feed, Icon, Comment } from 'semantic-ui-react';

type Props = {
  author: string,
  date: string,
  label: string,
  isLast: string,
};

/**
 * The component that displays a comment.
 */
export default function Activity(props: Props): React.Element<*> {

  const {
    author,
    date,
    label,
    isLast,
  } = props;

  const styles = reactCSS({
    'default': {
      feedContainer: {
        position: 'relative',
      },
      feedIcon: {
        fontSize: 10,
        marginTop: 20,
        marginLeft: -20,
        position: 'absolute',
      },
      commentGroup : {
        paddingTop: 0,
      },
    },
  });

  return (
    <Comment>
      <Feed>
        <Feed.Event>
          <Feed.Label style={styles.feedContainer}>
            <img src='http://semantic-ui.com/images/avatar/small/elliot.jpg' />
            <Icon name="circle" color="green" style={styles.feedIcon} />
          </Feed.Label>
          <Feed.Content>
            <Feed.Summary>
              <div>{author}{label}</div>
              <Feed.Date>{date}</Feed.Date>
            </Feed.Summary>
          </Feed.Content>
        </Feed.Event>
      </Feed>
      {isLast !== "true"? <Comment.Group style={styles.commentGroup}/>: null}
    </Comment>
  );
}
