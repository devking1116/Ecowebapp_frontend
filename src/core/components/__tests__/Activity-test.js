import React from 'react';
import renderer from 'react-test-renderer';

import Activity from '../Activity';

const testActivity = {
  author: 'User 0',
  date: '1/1/2016',
  label: 'commented on this post',
};

test('it displays the given comment.', () => {
  const component = renderer.create(
    <Activity {...testActivity} />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
