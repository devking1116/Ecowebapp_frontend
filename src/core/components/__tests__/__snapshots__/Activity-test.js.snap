exports[`test it displays the given comment. 1`] = `
<div
  className="comment">
  <div
    className="ui feed">
    <div
      className="event">
      <div
        className="label"
        style={
          Object {
            "position": "relative",
          }
        }>
        <img
          src="http://semantic-ui.com/images/avatar/small/elliot.jpg" />
        <i
          aria-hidden="true"
          className="green circle icon"
          style={
            Object {
              "fontSize": 10,
              "marginLeft": -20,
              "marginTop": 20,
              "position": "absolute",
            }
          } />
      </div>
      <div
        className="content">
        <div
          className="summary">
          <div>
            User 0
            commented on this post
          </div>
          <div
            className="date">
            1/1/2016
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui comments"
    style={
      Object {
        "paddingTop": 0,
      }
    } />
</div>
`;
